gnomad_qc/v4/sample_qc/relatedness.py

```diff
@@ -0,0 +1,251 @@
+                    project_meta_ht.exclude,
+                    chr20_mean_dp=sex_ht[project_meta_ht.key].chr20_mean_dp,
+                    filtered=hl.or_else(
+                        hl.len(
+                            hard_filtered_samples_ht[project_meta_ht.key].hard_filters
+                        )
+                        > 0,
+                        False,
+                    ),
+                )
+
+            from gnomad_qc.v3.resources.meta import project_meta as project_meta_v3
+            from gnomad_qc.v4.resources.meta import project_meta as project_meta_v4
+            from gnomad_qc.v3.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v3,
+            )
+            from gnomad_qc.v4.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v4,
+            )
+            from gnomad_qc.v3.resources.sample_qc import sex as sex_v3
+            from gnomad_qc.v4.resources.sample_qc import sex as sex_v4
+
+            annotated_project_meta_v3_ht = annotate_project_meta_ht(
+                project_meta_v3.ht(), sex_v3.ht(), hard_filtered_samples_v3.ht()
+            )
+            annotated_project_meta_v4_ht = annotate_project_meta_ht(
+                project_meta_v4.ht(), sex_v4.ht, hard_filtered_samples_v4.ht()
+            )
+
+            rank_ht = annotated_project_meta_v3_ht.union(
+                annotated_project_meta_v4_ht, unify=True
+            )
+
+            rank_ht = rank_ht.order_by(
+                rank_ht.filtered,
+                hl.desc(rank_ht.releasable & ~rank_ht.exclude),
+                ~hl.is_defined(  # Favor v3 samples.
+                    annotated_project_meta_v3_ht[rank_ht.s]
+                ),
+                hl.desc(rank_ht.chr20_mean_dp),
+            ).add_index(name="rank")
+
+            rank_ht = rank_ht.key_by(rank_ht.s)
+            rank_ht = rank_ht.select(rank_ht.filtered, rank_ht.rank)
+
+            rank_ht.write(samples_rankings.path)
+            rank_ht = samples_rankings.ht()
+
+            filtered_samples = hl.literal(
+                rank_ht.aggregate(
+                    hl.agg.filter(rank_ht.filtered, hl.agg.collect_as_set(rank_ht.s))
+                )
+            )
+
+            samples_to_drop = compute_related_samples_to_drop(
+                relatedness.ht(),
+                rank_ht,
+                args.second_degree_kin_cutoff,
+                filtered_samples=filtered_samples,
+            )
+            samples_to_drop = samples_to_drop.key_by(samples_to_drop.s)
+            samples_to_drop.write(related_samples_to_drop.path)
+
+    finally:
+        logger.info("Copying hail log to logging bucket...")
+        hl.copy_log(get_logging_path("relatedness"))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--test", help="Use a test MatrixTableResource as input.", action="store_true"
```

Comment:
Does that mean the semantics of `test` for `get_predetermined_qc` will also change? Or do you think that using the test subset for `get_predetermined_qc` isn't useful for testing this relatedness script?

I was assuming that any test dataset had already been prepared before running this script. Certainly no problem to add chromosome filtering here, but just curious what your usual strategy is to test the various stages.

Reply:
I'm thinking I might change the QC MT test to use the test dataset instead (can then also be used in the pop inference), so actually this will still work as is.